src/libultra/io/sirawdma.c

```diff
@@ -1,24 +1,24 @@
 #include "global.h"
 
 s32 __osSiRawStartDma(s32 direction, void* dramAddr) {
-    if ((*(u32*)0xA4800018 & 0x3) != 0) {
+    if (HW_REG(SI_STATUS_REG, u32) & (SI_STATUS_DMA_BUSY | SI_STATUS_IO_READ_BUSY)) {
         return -1;
     }
 
-    if (direction == 1) {
-        osWritebackDCache(dramAddr, 64);
+    if (direction == OS_WRITE) {
+        osWritebackDCache(dramAddr, 0x40);
```

Comment:
`0x40` is the PIF RAM size. I don't think we have a define for this right now but maybe worthwhile.

Reply:
Where should such a define go?

Reply:
hardware.h next to PIF_RAM_START seems like a good place

Reply:
So PIF_RAM_SIZE?

Reply:
Yeah sounds good